frontend/src/components/problem/Selector.tsx

```diff
@@ -146,4 +152,66 @@ function ProblemSelector(props: ProblemSelectorProps) {
   );
 }
 
-export default ProblemSelector;
+export function TagSelector(props: TagSelectorProps) {
+  const { tags, selectedTags, onSelect } = props;
```

Comment:
I realized this is tough since there's small differences here and there (such as with the type of problem vs. tag) but are there any options to reduce the amount of code duplication here? One thing off the top of my head is something like: passing the type in the list of items + a function taking in an object of that type and returning the desired html/component (is this similar at all to the createButtonLink thing we have?), maybe also a function that takes in a search query + a problem and returns a boolean to indicate whether it's filtered or not

Maybe another thing could also be to extract out the ref/useEffect logic (possibly into a custom hook) and other non-html logic

Reply:
Given that the `ProblemSelector` is a bit different (mostly given the inner difficulty widget), I'm thinking that these could be separate functions, but any future selectors could use the `TagSelector` as a general solution (with `ProblemSelector` still separate). Thoughts @alankbi?

Reply:
Sounds good - my only other suggestion would be if lines 164-173 in Selector.tsx could be made into a custom useX hook (since I think all the logic other than setShowTags/setShowProblems is reused) - that should hopefully be very easy/quick to do tho

Reply:
Done! ✅ 

This implementation was a bit different than the others because the `useClickOutside` hook did not return anything, but this seemed to work fine. 😃